addon/src/decorators/withHTML.js

```diff
@@ -14,6 +23,16 @@ export const withHTML = makeDecorator({
       if (parameters.removeEmptyComments) {
         html = html.replace(/<!--\s*-->/g, '');
       }
+      if (parameters.removeComments) {
+        const { removeComments } = parameters;
+        if (typeof removeComments === 'boolean') {
+          // Will remove all HTML comments
+          html = html.replace(/<!--(.*?)-->/g, '');
+        } else if (isValidRegExp(removeComments)) {
+          // Will remove only HTML comments matching the expression provided
+          html = html.replace(removeComments, '');
```

Comment:
This would allow removing any string from the HTML, not just comments. The regex should only target the content of the comment.

Reply:
Hi @jeanfredrik just to follow=up, are you referring to:
```js
html = html.replace(/<!--(.*?)-->/g, '');
```
or
```js
html = html.replace(removeComments, '');
```
please keep in mind that in the last one, `removeComments` is the regexp provided by the user.

Or perhaps we're talking the conditional is the issue?

Reply:
I mean the second part. You have to explicitly provide the comment open/close tags in the regex (like in your example) which I think is a but confusing. If I set `{ removeComments: /.lit.*?/g }` it would remove more things than just comments. I think it would be safer to have `{ removeComments: ".lit.*?" }` where the string is wrapped in the comment tags and made into a regex.